File: DiscoDruid3/app/src/main/java/com/andruid/magic/discodruid/viewholder/ArtistViewHolder.java
package com.andruid.magic.discodruid.viewholder;

import com.andruid.magic.discodruid.databinding.LayoutArtistBinding;
import com.andruid.magic.discodruid.model.Artist;

import androidx.recyclerview.widget.RecyclerView;

public class ArtistViewHolder extends RecyclerView.ViewHolder {
    private LayoutArtistBinding binding;

    public ArtistViewHolder(LayoutArtistBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(Artist artist){
        binding.setArtist(artist);
        binding.executePendingBindings();
    }
}

File: OldAttemptDiscoDruid/app/src/main/java/com/andruid/magic/discodruid/ui/viewholder/ArtistViewHolder.kt
package com.andruid.magic.discodruid.ui.viewholder

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.andruid.magic.discodruid.R
import com.andruid.magic.discodruid.databinding.LayoutArtistBinding
import com.andruid.magic.medialoader.model.Artist

class ArtistViewHolder(private val binding: LayoutArtistBinding) : RecyclerView.ViewHolder(binding.root) {
    companion object {
        @JvmStatic
        fun from(parent: ViewGroup): ArtistViewHolder {
            val inflater = LayoutInflater.from(parent.context)
            val binding = DataBindingUtil.inflate<LayoutArtistBinding>(inflater, R.layout.layout_artist,
                parent, false)
            return ArtistViewHolder(binding)
        }
    }

    fun bind(artist: Artist) {
        binding.artist = artist
        binding.executePendingBindings()
    }
}